codify/deployment.go

```diff
@@ -28,36 +28,40 @@ import (
 	"fmt"
 	"github.com/kris-nova/logger"
 	appsv1 "k8s.io/api/apps/v1"
+	"strings"
 	"text/template"
 	"time"
 )
 
 type Deployment struct {
-	i *appsv1.Deployment
+	*appsv1.Deployment
+	GoName string
 }
 
 func NewDeployment(obj *appsv1.Deployment) *Deployment {
 	obj.ObjectMeta = cleanObjectMeta(obj.ObjectMeta)
 	obj.Status = appsv1.DeploymentStatus{}
 	return &Deployment{
-		i: obj,
+		Deployment: obj,
+		GoName:     strings.ReplaceAll(obj.Name, "-", "_"),
 	}
 }
 
 func (k Deployment) Install() string {
-	l := Literal(k.i)
+	l := Literal(k.Deployment)
```

Comment:
see my not above about how we would now have to change every file to reference the embedded type instead of a generic field name